### PR TITLE
cmd,sink(ticdc): validate changefeed params and revise error message 

### LINKED ISSUE #1716: worker-count should has limit to set value
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

- Create changefeed --sink-uri='xxx/worker-count=0' 
image

- Create changefeed --sink-uri='xxx/worker-count=1000000000000000'
image

2. What did you expect to see?

- worker-count=0 is not allowed to set
- What is the max value for  worker-count? worker-count=1000000000000000 seems too large to allowed be to set.

3. What did you see instead?
Successfully to create changefeed with worker-count=0 or worker-count=1000000000000000
4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
       5.0.0-nightly
        ```

    - TiCDC version (execute `cdc version`):

        ```
        5.0.0-nightly
        ```

### LINKED ISSUE #1718: max-txn-row should has limit value
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

- Create or update  changefeed with set max-txn-rowbe to invalid value 0, -1
image

- 

2. What did you expect to see?
Failed to create or update changefeed.

3. What did you see instead?
Success to create or update changefeed. What's more, changefeed sync task seems still can move on. 
4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
       5.0.0-nightly
        ```

    - TiCDC version (execute `cdc version`):

        ```
        5.0.0-nightly
        ```

### LINKED ISSUE #1719: update --sort-engine should has limit since it only support memory and unified
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

- update --sort-engine with invalid value, such as null, u, memo...
image

- 

2. What did you expect to see?
Failed to update changefeed
3. What did you see instead?

4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
       5.0.0-nightly
        ```

    - TiCDC version (execute `cdc version`):

        ```
        5.0.0-nightly
        ```

### LINKED ISSUE #4472: Error message is not clear when pd address not found
### What did you do?

-  /cdc cli changefeed list --pd=http://upstream:2379
Error: context deadline exceeded
Usage:
  cdc cli changefeed list [flags]

### What did you expect to see?

such as: PD not found, please make sure PD address is correct...

### What did you see instead?

Error: context deadline exceeded

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(5.5 nightly
```

TiCDC version (execute `cdc version`):

```console
5.5 nightly
```

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve?

Issue Number: close #1716 close #1718 close #1719 close #4472

### Check List 

Tests 

 - Unit test

### Release note 

```release-note
Validate changefeed parameters on creation and revise error message
```

### CODE REVIEW

File: cdc/sink/mysql_params.go [hunk 1]
```diff
@@ -35,6 +35,8 @@ const (
 	// expose these two variables for redo log applier
 	DefaultWorkerCount = 16
 	DefaultMaxTxnRow   = 256
+	maxWorkerCount     = 1024
```
Comment (3AceShowHand) [inline cdc/sink/mysql_params.go:38]: how about prefix with `default` to uniform the naming convention?

can you add some comment about why chose these values.

File: pkg/cmd/cli/cli_changefeed_create.go [hunk 2]
```diff
@@ -227,6 +228,14 @@ func (o *createChangefeedOptions) completeCfg(ctx context.Context, cmd *cobra.Co
 		}
 	}
 
+	if o.commonChangefeedOptions.sortEngine != model.SortInMemory &&
```
Comment (3AceShowHand) [inline pkg/cmd/cli/cli_changefeed_create.go:231]: Is it possible be replaced by a `switch` ?

File: cdc/sink/mysql_params.go [hunk 3]
```diff
@@ -113,16 +115,32 @@ func parseSinkURIToParams(ctx context.Context, sinkURI *url.URL, opts map[string
 		if err != nil {
 			return nil, cerror.WrapError(cerror.ErrMySQLInvalidConfig, err)
 		}
-		if c > 0 {
-			params.workerCount = c
+		if c <= 0 {
+			return nil, cerror.WrapError(cerror.ErrMySQLInvalidConfig,
+				fmt.Errorf("invalid worker-count %d", c))
```
Comment (hi-rustin) [inline cdc/sink/mysql_params.go:120]:
```suggestion
				fmt.Errorf("invalid worker-count %d, which must be greater than 0", c))
```

File: cdc/sink/mysql_params.go [hunk 4]
```diff
@@ -113,16 +115,32 @@ func parseSinkURIToParams(ctx context.Context, sinkURI *url.URL, opts map[string
...
 	}
 	s = sinkURI.Query().Get("max-txn-row")
 	if s != "" {
 		c, err := strconv.Atoi(s)
 		if err != nil {
 			return nil, cerror.WrapError(cerror.ErrMySQLInvalidConfig, err)
 		}
+		if c <= 0 {
+			return nil, cerror.WrapError(cerror.ErrMySQLInvalidConfig,
+				fmt.Errorf("invalid max-txn-row %d", c))
```
Comment (hi-rustin) [inline cdc/sink/mysql_params.go:137]: ditto

**Review by 3AceShowHand — APPROVED** [review]

**Review by hi-rustin — APPROVED** [review]
Great 👍

File: cdc/sink/mysql_params.go (hunk 1 above)
Comment (hi-rustin) [reply cdc/sink/mysql_params.go:38]: This is not the default value.

**Pull request merged**